server/text_generation_server/layers/gptq/__init__.py

```diff
@@ -350,16 +352,16 @@ def get_weights_row(self, weights: Weights, prefix: str):
             else:
                 log_once(logger.info, f"Using exllama kernels v{HAS_EXLLAMA}")
 
-        if use_exllama and self.groupsize != -1:
+        if not desc_act and self.groupsize != -1:
```

Comment:
use_exllama is False since exllama only support cuda. but ipex quantization runtime kernel also implement similar logic like exllama. so we need such sharded logic for qzeros/scale/g_idx if desc_act is False as well.

Reply:
Isn't keeping `use_exllama` and simply fixing the TP (with - g_idx[0]) in the conditional to fix the issues on IPEX ?

Reply:
https://github.com/huggingface/text-generation-inference/blob/main/server/text_generation_server/layers/gptq/__init__.py#L134,this line set the use_exllama to false, since in intel platform, exllama is not installed.